app/controllers/application_controller/filter.rb

```diff
@@ -1,10 +1,13 @@
 # Filter/search/expression methods included in application.rb
+require_relative '../../helpers/miq_expression/filter_subst'
```

Comment:
Looking at this and reading Jason's comments I agree that this code would be better suited as a module that gets mixed in to MiqExpression as class methods. But it could also be taken further by making them instance methods too. Maybe set the substituted expression an a new accessor or something like that.